spatialos-sdk/examples/project-example/main.rs

```diff
@@ -45,24 +45,83 @@ fn main() {
 
 fn logic_loop(c: &mut WorkerConnection) {
     let mut counter = 0;
+    let mut entity_id: Option<EntityId> = None;
 
     loop {
         let ops = c.get_op_list(0);
 
+        // Move entity (if it exists).
+        if let Some(ref entity_id) = entity_id {
+            // Counter == 20 is one full circle.
+            let angle = (counter as f64) * (f64::consts::PI * 2.0 / 20.0);
+            c.send_component_update::<improbable::Position>(
+                entity_id.clone(),
+                improbable::PositionUpdate {
+                    coords: Some(improbable::Coordinates {
+                        x: angle.sin() * 10.0,
+                        y: 0.0,
+                        z: angle.cos() * 10.0,
+                    }),
+                },
+                UpdateParameters { loopback: true },
+            );
+            println!("Sending component update for improbable::Position to entity {:?}.", entity_id);
+        }
+
         // Process ops.
         for op in &ops {
-            println!("Received op: {:?}", op);
+            if let WorkerOp::Metrics(_) = op {
+                println!("Received metrics.");
+            } else {
+                println!("Received op: {:?}", op);
+            }
             match op {
+                WorkerOp::ReserveEntityIdsResponse(response) => {
+                    if let StatusCode::Success(response_data) = response.status_code {
+                        let mut entity = Entity::new();
```

Comment:
I didn't factor this out into a separate method, but I did move this out of the `match` block to it's own section of the function, which I think makes the setup logic a bit cleaner. I'm hesitant to pull the entity creation logic out into its own function since it's only used in one place, but if you'd still prefer it to be separated then I can go ahead and do that 🙂

Reply:
Yup it looks fine now - it was the nesting which I disliked 👍